res/controllers/Hercules-DJControl-Inpulse-300-script.js

```diff
@@ -0,0 +1,231 @@
+// DJControl_Inpulse_300_script.js
+//
+// ***************************************************************************
+// * Mixxx mapping script file for the Hercules DJControl Inpulse 300.
+// * Author: DJ Phatso, contributions by Kerrick Staley
+// * Version 1.2 (December 2019)
+// * Forum: https://www.mixxx.org/forums/viewtopic.php?f=7&t=12599
+// * Wiki: https://mixxx.org/wiki/doku.php/hercules_djcontrol_inpulse_300
+//
+// Changes to v1.2
+// - Code cleanup.
+//
+// Changes to v1.1
+// - Fix seek-to-start and cue-master behavior.
+// - Tweak scratch, seek, and bend behavior.
+// - Controller knob/slider values are queried on startup, so MIXXX is synced.
+// - Fixed vinyl button behavior the first time it's pressed.
+//
+// v1.0 : Original forum release
+//
+// TODO: Functions that could be implemented to the script:
+//
+// * ROLL: Keep SLIP active (if already enabled) when exiting from rolls
+//
+// * SLICER/SLICER LOOP
+//
+// * TONEPLAY
+//
+// * FX:
+//	- Potentially use 1 FX rack for FX pads and another for the Controlled FX
+//  - See how to preselect effects for a rack
+// * Fix behavior when adjusting tempo slider after pressing [Sync] (tempo adjustment should be relative, not absolute).
```

Comment:
Have a look at: https://www.mixxx.org/wiki/doku.php/midi_scripting#soft-takeover

Reply:
So if I understand correctly, for this to work, I have to remove the pitch fader assigments from the XML and map them to the script file ?

Reply:
Yes, I think so.